packages/astro/src/@types/astro.ts

```diff
@@ -51,11 +51,13 @@ export interface AstroGlobal extends AstroGlobalPartial {
 }
 
 export interface AstroGlobalPartial {
-	fetchContent<T = any>(globStr: string): Promise<FetchContentResult<T>[]>;
 	/**
 	 * @deprecated since version 0.24. See the {@link https://astro.build/deprecated/resolve upgrade guide} for more details.
 	 */
 	resolve: (path: string) => string;
+	/** @deprecated Use `Astro.glob()` instead. */
+	fetchContent(globStr: string): Promise<any[]>;
+	glob<T = any>(globStr: string): Promise<T[]>;
```

Comment:
We can narrow these a little based on the file extension for a better DX! This would keep folks from needing to wrap types in `{frontmatter: { ... }}` for globs ending in `.md`, but still keeps it open for fully custom types if no extension matches.

```suggestion
	glob<T extends Record<string, any>>(globStr: `${any}.md`): Promise<MarkdownGlobResult<T>[]>;
	glob<T extends Record<string, any>>(globStr: `${any}.astro`): Promise<AstroGlobResult[]>;
	glob<T extends Record<string, any>>(globStr: string): Promise<T[]>;
```

Assuming these types are also added to this file.

```ts
export interface MarkdownGlobResult<T extends Record<string, any>> extends MarkdownInstance {
	frontmatter: T;
}
export interface AstroGlobResult {
	default: AstroComponentFactory;
}
```

Reply:
TS is 🤩